packages/swapper/src/swappers/PortalsSwapper/getPortalsTradeQuote/getPortalsTradeQuote.ts

```diff
@@ -86,124 +91,171 @@ export async function getPortalsTradeQuote(
 
   if (!sendAddress) return Err(makeSwapErrorRight({ message: 'missing sendAddress' }))
 
-  try {
-    const portalsNetwork = chainIdToPortalsNetwork[chainId as KnownChainIds]
+  const portalsNetwork = chainIdToPortalsNetwork[chainId as KnownChainIds]
 
-    if (!portalsNetwork) {
+  if (!portalsNetwork) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported ChainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: input.chainId },
+      }),
+    )
+  }
+
+  const sellAssetAddress = isNativeEvmAsset(sellAsset.assetId)
+    ? zeroAddress
+    : fromAssetId(sellAsset.assetId).assetReference
+  const buyAssetAddress = isNativeEvmAsset(buyAsset.assetId)
+    ? zeroAddress
+    : fromAssetId(buyAsset.assetId).assetReference
+
+  const inputToken = `${portalsNetwork}:${sellAssetAddress}`
+  const outputToken = `${portalsNetwork}:${buyAssetAddress}`
+
+  const maybePortalsTradeOrderResponse = await fetchPortalsTradeOrder({
+    sender: sendAddress,
+    inputToken,
+    outputToken,
+    inputAmount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    slippageTolerancePercentage: userSlippageTolerancePercentageDecimalOrDefault,
+    partner: getTreasuryAddressFromChainId(sellAsset.chainId),
+    feePercentage: affiliateBpsPercentage,
+    validate: true,
+    swapperConfig,
+  })
+    .then(res => Ok(res))
+    .catch(async err => {
+      if (err instanceof PortalsError) {
+        // We assume a PortalsError was thrown because the slippage tolerance was too high during simulation
+        // So we attempt another (failing) call with autoslippage which will give us the actual expected slippage
+        const portalsExpectedSlippage = await fetchPortalsTradeOrder({
+          sender: sendAddress,
+          inputToken,
+          outputToken,
+          inputAmount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          autoSlippage: true,
+          partner: getTreasuryAddressFromChainId(sellAsset.chainId),
+          feePercentage: affiliateBpsPercentage,
+          validate: true,
+          swapperConfig,
+        })
+          // This should never happen but could in very rare cases if original call failed on slippage slightly over 2.5% but this one succeeds on slightly under 2.5%
+          .then(res => res.context.slippageTolerancePercentage)
+          .catch(err => (err as PortalsError).message.match(/Expected slippage is (.*?)%/)![1])
```

Comment:
🧠

Reply:
[question]: Is there ever a case where the error message doesnt match this regexp? Would it be worth handling this case?

Reply:
With autoslippage, there *should* not AFAIK.
There are only 2 errors that can be returned from `/portal` calls that I'm aware of:
- Auto-slippage exceeded (this guy) which only happens when auto-slippage is enabled (no explicit slippage passed)
- Execution reverted

I think the latter could indeed be a valid issue: since the original call is done with validation (simulation) enabled, there could be an `execution reverted` error which is not a result of slippage but just a non-tradeable pair that fails. If that's the case, this call with auto-slippage would yield the same `execution reverted` and an exception would be thrown because of the `![1]` non-null assertion here. 

Tackled in 9de0393efc - IMO this is edge-case enough (I'm not even aware of pairs this would happen with) not to warrant explicit copies, but we can always add a generic execution reverted with no slippage interpolation params if we see these often and they're ugly

Reply:
As discussed, brought back the try/catch as a last resort paranoia @woodenfurniture 
fc20fa4be2f279f05ed666693ccb839e4f1163d9